src/Display/Modal/Modal.test.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import Modal from '.';
+
+import * as renderer from 'react-test-renderer';
+import '@testing-library/jest-dom/extend-expect';
+import { fireEvent, render } from '@testing-library/react';
+
+const props = {
+  title: 'Modal Title',
+  content: 'Modal Content',
+  footer: 'Modal Footer',
+  onClose: jest.fn(),
+};
+
+const OpenedModal = (
+  <Modal
+    title={props.title}
+    isVisible={true}
+    footer={[<button key="button">{props.footer}</button>]}
+    onClose={props.onClose}
+  >
+    <p>{props.content}</p>
+  </Modal>
+);
+
+function setupModal(isVisible: boolean) {
```

Comment:
hey @franciscrispin 

When referencing storybook, I saw a few more props that this modal can consume.

I believe that there is merit in testing the `size`, `hideHeader`, and `hideContentArea` props. 
I lack knowledge in testing for the `centering` prop -- I will need your advice on that.
(if you are also not aware, lets pair program this!)

Reply:
Hmm yeah I was considering those. My initial idea was to make sure every single prop was tested. I'm not so sure whether that's a good idea cause I would probably end up testing implementation details like checking for whether `align-items: center` (`centering`) exists or whether `background: transparent` (`hideContentArea`) exists. 

I recalled this from the `react-testing-library` [docs](https://testing-library.com/docs/intro) which seems to imply that doing the above is unfavourable. 
```
As a part of this goal, you want your tests to avoid including implementation details so refactors of your components (changes to implementation but not functionality) don't break your tests and slow you and your team down.
```

I'm not sure what's the right approach for this also so if you have any thoughts please do share them.

Reply:
> As a part of this goal, you want your tests to avoid including implementation details so refactors of your components (changes to implementation but not functionality) don't break your tests and slow you and your team down.

I find that line above interesting and really changes the way we approach tests. 
I do agree that checking for implementation details is not healthy. 

I have a possible solution to test for `hideHeader` and `hideContentArea`. 
What if we test using the `has text` thing?
That way, _we are testing that the header text does not appear_. 

What do you think?

Reply:
I realised `hideContentArea` might be an unnecessary prop which may be deleted so I left out tests for that. Added test for `size`, `centering` and `hideHeader`. For the first two, I had to test the styling itself for it to work. Don't think there's another way around it.